tests/components/sensoterra/test_config_flow.py

```diff
@@ -0,0 +1,148 @@
+    )
+
+    hass.data["core.uuid"] = HASS_UUID
+
+    with patch(
+        "sensoterra.customerapi.CustomerApi.get_token",
+        side_effect=StInvalidAuth("Invalid credentials"),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_EMAIL: API_EMAIL,
+                CONF_PASSWORD: API_PASSWORD,
+            },
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+    # Make sure the config flow tests finish with either an
+    # FlowResultType.CREATE_ENTRY or FlowResultType.ABORT so
+    # we can show the config flow is able to recover from an error.
+    with patch(
+        "sensoterra.customerapi.CustomerApi.get_token",
+        return_value=API_ACCESS_TOKEN,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_EMAIL: API_EMAIL,
+                CONF_PASSWORD: API_PASSWORD,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == API_EMAIL
+    assert result["data"] == {
+        CONF_TOKEN: API_ACCESS_TOKEN,
+        CONF_EMAIL: API_EMAIL,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    hass.data["core.uuid"] = HASS_UUID
+
+    with patch(
+        "sensoterra.customerapi.CustomerApi.get_token",
+        side_effect=StTimeout,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_EMAIL: API_EMAIL,
+                CONF_PASSWORD: API_PASSWORD,
+            },
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    # Make sure the config flow tests finish with either an
+    # FlowResultType.CREATE_ENTRY or FlowResultType.ABORT so
+    # we can show the config flow is able to recover from an error.
+    with patch(
+        "sensoterra.customerapi.CustomerApi.get_token",
+        return_value=API_ACCESS_TOKEN,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_EMAIL: API_EMAIL,
+                CONF_PASSWORD: API_PASSWORD,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == API_EMAIL
+    assert result["data"] == {
+        CONF_EMAIL: API_EMAIL,
+        CONF_TOKEN: API_ACCESS_TOKEN,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
```

Comment:
These 2 tests can be combined using `pytest.mark.parametrize` if you do a global search in the code you can find examples of this for the config flow exception tests